techmatters/hrm
# Move conversationMedia out of the contacts JSON blob into it's own table

## Description
This PR:
- Adds a migration to create `ConversationMedias` table.
- Refactors contact logic to create new entries on the above table if there are conversation media entries in the payload.
- Refactors all queries to include conversation media relations where a contact is retrieved.
- Refactors the contact jobs system to contemplate the above changes.
- Added migration to convert existing information to the new table - the legacy media is not deleted.

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-2022)

### Verification steps
- Start development server - this branch for HRM (`npm run ssm:local & npm start`), master for Serverless (`npm run start:local`) and Flex (`npm run dev:local:serverless`).
- Start a new chat contact and submit it.
- Wait for the transcript to be exported.
- Confirm  that in all places where a contact can be accessed, the transcript is too (search contacts, search cases, cases list).
- Confirm that in the DB the records are as expected:
  - `Contact` -> `rawJson` does not contains the `conversationMedia` anymore.
  - A new S3 type record exists in `ConversationMedia` for the above contact, with the appropriate information (bucket and location).
  - The `ContactJob` for the transcript export is appropriately marked as complete.
